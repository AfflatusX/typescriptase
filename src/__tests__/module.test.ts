/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/tests.ts::moduleTest>>
 * BESPOKE<<imports, beforeAll, afterAll, beforeEach, afterEach, complexModule, contentModule, emptyModule, importModule>>
 * SIGNED<<YGhzEB4FEV+2OB2EBuQHTpU5p4kRmZ3xJmsDfuDVi8zHX/eOZHc34b2qlmMvaCnwX3rjgwnzWJ8buPgca1nqEA==>>
 */

/* BESPOKE START <<imports>> */
import { complexModule, contentModule, emptyModule, importModule } from "./examples";
/* BESPOKE END <<imports>> */

beforeAll(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeAll>> */
    /* BESPOKE END <<beforeAll>> */
  },
);

afterAll(
  async (): Promise<void> => {
    /* BESPOKE START <<afterAll>> */
    /* BESPOKE END <<afterAll>> */
  },
);

beforeEach(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeEach>> */
    /* BESPOKE END <<beforeEach>> */
  },
);

afterEach(
  async (): Promise<void> => {
    /* BESPOKE START <<afterEach>> */
    /* BESPOKE END <<afterEach>> */
  },
);

test(
  "complexModule",
  async (): Promise<void> => {
    /* BESPOKE START <<complexModule>> */
    expect(complexModule.print({
      name: "complexModule",
      path: "src/__tests__/module.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<complexModule>> */
  },
);

test(
  "contentModule",
  async (): Promise<void> => {
    /* BESPOKE START <<contentModule>> */
    expect(contentModule.print({
      name: "contentModule",
      path: "src/__tests__/module.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<contentModule>> */
  },
);

test(
  "emptyModule",
  async (): Promise<void> => {
    /* BESPOKE START <<emptyModule>> */
    expect(emptyModule.print({
      name: "emptyModule",
      path: "src/__tests__/module.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<emptyModule>> */
  },
);

test(
  "importModule",
  async (): Promise<void> => {
    /* BESPOKE START <<importModule>> */
    expect(importModule.print({
      name: "importModule",
      path: "src/__tests__/module.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<importModule>> */
  },
);
