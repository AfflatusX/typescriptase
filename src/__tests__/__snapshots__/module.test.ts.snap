// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complexModule 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<src/__tests__/module.test.ts::complexModule>>
 * BESPOKE<<TEST1, fn3, methodInstance3Bespoke, methodStatic3Bespoke, 3tESt3>>
 * SIGNED<<s0gzV06h8Hjkjkb2IBwcjDlftkeYFMfzpRk4VZCNd4DdPuoWdnoc1tOCmuuhuHIA41To35XpNAsPr4+qnhooQg==>>
 */

import \\"./p\\";
import \\"aqe\\";
import \\"ZZZ\\";

import {
  B,
  c,
} from \\"a\\";
import * as aaa from \\"BbB\\";
import ooo from \\"nnn\\";
import xxx from \\"www\\";
import * as zzz from \\"YYY\\";

import {
  g,
  r,
  z,
} from \\"./a\\";
import fff from \\"./ddd\\";
import * as mmm from \\"./mmm\\";
import {
  A,
  wA,
} from \\"./R\\";

export class MyClass2<T, V> extends MyClass1 {

  /* BESPOKE START <<TEST1>> */
  /* BESPOKE END <<TEST1>> */

  protected readonly propertyInstance2: number;

  protected readonly static propertyStatic2: number;

  protected abstract methodInstance2<T, V>(
  ): void;

  protected static abstract methodStatic2<T, V>(
  ): void;
}

interface If1<T> {
}

export async function function3(
  var1: string,
  var2: object = {},
): Promise<string[]> {
  /* BESPOKE START <<fn3>> */
  /* BESPOKE END <<fn3>> */
}

let variable3: number | null | undefined = 1;

export interface If2<T, V> {
  item: string;
}

type Ty1<T> = number;

export type Ty2<T, V> = string | null;

export type Ty3 = Ty2 | Ty1;

class MyClass3 extends MyClass1 implements MyInterface1, MyInterface2 {

  private propertyInstance3: string | null;

  private static propertyStatic3: string | null;

  private async methodInstance3(
  ): void {
    /* BESPOKE START <<methodInstance3Bespoke>> */
    /* BESPOKE END <<methodInstance3Bespoke>> */
  }

  private static async methodStatic3(
  ): void {
    /* BESPOKE START <<methodStatic3Bespoke>> */
    /* BESPOKE END <<methodStatic3Bespoke>> */
  }
}

export async function function2<T, V>(
): Promise<string> {
}

export const variable1: string;

/* BESPOKE START <<3tESt3>> */
/* BESPOKE END <<3tESt3>> */

const variable2: string | undefined = \\"TEST\\";

export interface If3 {
  test: null | string;
  test2: If2;
}

function function1<T>(
): void {
}
"
`;

exports[`contentModule 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<src/__tests__/module.test.ts::contentModule>>
 * BESPOKE<<TEST1, test2, fn3>>
 * SIGNED<<gyZYU9ZfO/lD+zY23MVW2OpdXbv//9gvjiok8CLQ3bqm6uRUww6LbU0vI6YyxofW1LLSrCSSnbeNqJiWBf9+wA==>>
 */

export abstract class MyClass1<T> {

  public propertyInstance1: string = \\"MYVAR\\";

  public static propertyStatic1: string = \\"MYVAR\\";

  public async methodInstance1<T>(
    var1: string,
    var2: object = {},
  ): Promise<string[]> {
  }

  public static async methodStatic1<T>(
    var1: string,
    var2: object = {},
  ): Promise<string[]> {
  }
}

type Ty1<T> = number;

/* BESPOKE START <<TEST1>> */
/* BESPOKE END <<TEST1>> */

interface If1<T> {
}

function function1<T>(
): void {
}

export const variable1: string;

export type Ty2<T, V> = string | null;

export async function function2<T, V>(
): Promise<string> {
}

export interface If3 {
  test: null | string;
  test2: If2;
}

const variable2: string | undefined = \\"TEST\\";

let variable3: number | null | undefined = 1;

/* BESPOKE START <<test2>> */
/* BESPOKE END <<test2>> */

export async function function3(
  var1: string,
  var2: object = {},
): Promise<string[]> {
  /* BESPOKE START <<fn3>> */
  /* BESPOKE END <<fn3>> */
}

export interface If2<T, V> {
  item: string;
}

export type Ty3 = Ty2 | Ty1;
"
`;

exports[`emptyModule 1`] = `
"/**
 * This file is fully generated; do not manually edit.
 *
 * SOURCE<<src/__tests__/module.test.ts::emptyModule>>
 * SIGNED<<z4PhNX7vuL3xVChQ1m2AB9Yg5AULVxXcg/SpIdNs6c5H0NE8XYXysP+DGNKHfuwvY7kxvUdBeoGlODJ6+SfaPg==>>
 */
"
`;

exports[`importModule 1`] = `
"/**
 * This file is fully generated; do not manually edit.
 *
 * SOURCE<<src/__tests__/module.test.ts::importModule>>
 * SIGNED<<nuskwMYvDowcShGj7ARGsTL7b/BBCoDcLpaOivMO6l3A9oECfnr089sFlub49/+ApuhHjOHuo+PyeRSzJJ0JXw==>>
 */

import \\"./p\\";
import \\"aqe\\";
import \\"ZZZ\\";

import {
  B,
  c,
} from \\"a\\";
import * as aaa from \\"BbB\\";
import ooo from \\"nnn\\";
import xxx from \\"www\\";
import * as zzz from \\"YYY\\";

import {
  g,
  r,
  z,
} from \\"./a\\";
import fff from \\"./ddd\\";
import * as mmm from \\"./mmm\\";
import {
  A,
  wA,
} from \\"./R\\";
"
`;
